### PR TITLE
Use cloud-resource operator to provision external postgers in codeready- INTLY-3146

### PR DESCRIPTION
**Pull request by austincunningham**

### CODE REVIEW

File: pkg/controller/installation/products/codeready/reconciler.go [hunk 1]
```diff
@@ -459,6 +481,67 @@ func (r *Reconciler) createCheCluster(ctx context.Context, kcCfg *config.RHSSO,
...
+	if err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: secRef.Name, Namespace: secRef.Namespace}, credSec); err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve credential secret for %s", postgresCredName)
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      "postgres",
+		},
+	}
+	fmt.Println("austin", deployment.Spec.Template.Spec.Containers[0].Env)
```
Comment (aidenkeating) [inline pkg/controller/installation/products/codeready/reconciler.go]: do we need this?

File: pkg/controller/installation/products/codeready/reconciler.go [hunk 2]
```diff
@@ -163,8 +165,11 @@ func (r *Reconciler) reconcilePostgresSecret(ctx context.Context, serverClient p
 			postgresqlSecret.Data["POSTGRES_PASSWORD"] = []byte(env.Value)
 		case "POSTGRESQL_DATABASE":
 			postgresqlSecret.Data["POSTGRES_DATABASE"] = []byte(env.Value)
+		case "POSTGRESQL_ADMIN_PASSWORD":
+			postgresqlSecret.Data["POSTGRES_ADMIN_PASSWORD"] = []byte(env.Value)
 		}
 	}
+	fmt.Println("austin", deployment.Spec.Template.Spec.Containers[0].Env)
```
Comment (aidenkeating) [inline pkg/controller/installation/products/codeready/reconciler.go]: can we remove this

**Comment by ciaranRoche** [comment]
@aidenkeating @austincunningham Just noticed, on cleanup of the intly installation, we are not cleaning up the postgres instance as it is in the intly namespace. Do we want to add a owner ref or finalizer to postgres resource.

File: pkg/controller/installation/products/codeready/reconciler.go (hunk 1 above)
Comment (austincunningham) [reply pkg/controller/installation/products/codeready/reconciler.go]: no

File: pkg/controller/installation/products/codeready/reconciler.go (hunk 2 above)
Comment (austincunningham) [reply pkg/controller/installation/products/codeready/reconciler.go]: yes

**Pull request merged**